devicetypes/samsungsds/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy

```diff
@@ -0,0 +1,402 @@
+/**
+ *  Samsung Smart Doorlock
+ *
+ *  Copyright 2018 Samsung SDS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+import physicalgraph.zigbee.zcl.DataType
+
+
+metadata {
+	definition (name: "Samsung Smart Doorlock", namespace: "Samsung SDS", author: "kyun.park", mnmn: "SmartThings", vid: "generic-lock") {
+		capability "Actuator"
+		capability "Lock"
+		capability "Battery"
+		capability "Configuration"
+        capability "Health Check"
```

Comment:
Remember to update the indentation to use all tabs. It's easiest to do in an editor besides the IDE.